features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt

```diff
@@ -281,6 +288,58 @@ internal class RumSessionScope(
         )
     }
 
+    private fun handleTTIDEvent(
+        event: RumRawEvent.AppStartTTIDEvent,
+        datadogContext: DatadogContext,
+        writeScope: EventWriteScope,
+        writer: DataWriter<Any>
+    ) {
+        sdkCore.newRumEventWriteOperation(datadogContext, writeScope, writer) {
+            createTTIDVitalEvent(
+                event = event,
+                datadogContext = datadogContext
+            )
+        }.submit()
+
+        rumAppStartupTelemetryReporter.reportTTID(
+            info = event.info,
+            indexInSession = appStartIndex
+        )
+
+        appStartIndex++
+    }
+
+    private fun createTTIDVitalEvent(
+        event: RumRawEvent.AppStartTTIDEvent,
+        datadogContext: DatadogContext
+    ): VitalEvent {
+        val rumContext = getRumContext()
+
+        val hasReplay = childScope?.activeView?.viewId?.let {
```

Comment:
`viewId` can be taken simply from `RumContext`:

```suggestion
        val hasReplay = rumContext.viewId?.let {
```

Reply:
I don't think so. This is `RumSessionScope` and [getRumContext](https://github.com/DataDog/dd-sdk-android/blob/641c42dccd480ac8d714e9c8627a3dd2d4c6e16a/features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt#L182) doesn't fill `viewId` field. It is filled only [here](https://github.com/DataDog/dd-sdk-android/blob/641c42dccd480ac8d714e9c8627a3dd2d4c6e16a/features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt#L416) in `RumViewScope`.